domain/metrics.go

```diff
@@ -0,0 +1,126 @@
+package domain
+
+// Metric names that are allowed for processing by Forta
+const (
+	MetricClientDial  = "agent.client.dial"
+	MetricClientClose = "agent.client.close"
+
+	MetricStatusRunning     = "agent.status.running"
+	MetricStatusAttached    = "agent.status.attached"
+	MetricStatusInitialized = "agent.status.initialized"
+	MetricStatusStopping    = "agent.status.stopping"
+	MetricStatusActive      = "agent.status.active"
+	MetricStatusInactive    = "agent.status.inactive"
+
+	MetricActionUpdate      = "agent.action.update"
+	MetricActionRestart     = "agent.action.restart"
+	MetricActionSubscribe   = "agent.action.subscribe"
+	MetricActionUnsubscribe = "agent.action.unsubscribe"
+
+	MetricFailurePull               = "agent.failure.pull"
+	MetricFailureLaunch             = "agent.failure.launch"
+	MetricFailureStop               = "agent.failure.stop"
+	MetricFailureDial               = "agent.failure.dial"
+	MetricFailureInitialize         = "agent.failure.initialize"
+	MetricFailureInitializeResponse = "agent.failure.initialize.response"
+	MetricFailureInitializeValidate = "agent.failure.initialize.validate"
+	MetricFailureTooManyErrs        = "agent.failure.too-many-errs"
+
+	MetricHealthCheckAttempt = "agent.health.attempt"
+	MetricHealthCheckSuccess = "agent.health.success"
+	MetricHealthCheckError   = "agent.health.error"
+
+	MetricFinding       = "finding"
+	MetricTxRequest     = "tx.request"
+	MetricTxLatency     = "tx.latency"
+	MetricTxError       = "tx.error"
+	MetricTxSuccess     = "tx.success"
+	MetricTxDrop        = "tx.drop"
+	MetricTxBlockAge    = "tx.block.age"
+	MetricTxEventAge    = "tx.event.age"
+	MetricBlockBlockAge = "block.block.age"
+	MetricBlockEventAge = "block.event.age"
+	MetricBlockRequest  = "block.request"
+	MetricBlockLatency  = "block.latency"
+	MetricBlockError    = "block.error"
+	MetricBlockSuccess  = "block.success"
+	MetricBlockDrop     = "block.drop"
+
+	MetricJSONRPCLatency          = "jsonrpc.latency"
+	MetricJSONRPCRequest          = "jsonrpc.request"
+	MetricJSONRPCSuccess          = "jsonrpc.success"
+	MetricJSONRPCThrottled        = "jsonrpc.throttled"
+	MetricPublicAPIProxyLatency   = "publicapi.latency"
+	MetricPublicAPIProxyRequest   = "publicapi.request"
+	MetricPublicAPIProxySuccess   = "publicapi.success"
+	MetricPublicAPIProxyThrottled = "publicapi.throttled"
+	MetricFindingsDropped         = "findings.dropped"
+	MetricCombinerRequest         = "combiner.request"
+	MetricCombinerLatency         = "combiner.latency"
+	MetricCombinerError           = "combiner.error"
+	MetricCombinerSuccess         = "combiner.success"
+	MetricCombinerDrop            = "combiner.drop"
+)
+
+var (
+	Metrics = map[string]interface{}{
```

Comment:
Can we make this internal like `allMetrics` and then add an exposed method like `IsSupportedMetric()`?